game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -749,27 +768,67 @@ public JComponent getUiComponent() {
 
       @Override
       public void save(final SaveContext context) {
-        final String username = usernameField.getText();
-        context.setValue(usernameSetting, username.isEmpty() ? null : username.toCharArray());
+        // Only save when value changed
+        if (usernameField
+            .getText()
+            .equals(usernameSetting.getValue().map(String::new).orElse(""))) {
+          return;
+        }
+        try {
+          BackgroundTaskRunner.runInBackground(
+              "Fetching Login Token...",
+              () -> {
+                final NodeBbTokenGenerator tokenGenerator = new NodeBbTokenGenerator(forumUrl);
+                final Optional<Integer> oldUserId = uidSetting.getValue();
+                final Optional<char[]> oldToken = tokenSetting.getValue();
+                if (!usernameField.getText().isEmpty()) {
```

Comment:
Perhaps we want to use `isBlank()` instead of `isEmpty()`.

Do we currently already enforce a minimum length check on usename before the 'save' button is enabled? It might be the case that we've already guaranteed that there is a non-empty value that is of a minimum length before we hit this code which would make this validation redundant.

Side-comment, redundant client-side validation can generally be removed. If someone hacks it up so a button is enabled without validation, they can hack this. Server validation needs to be redundant in case a client hacks out the validation (or is just using straight-up `curl` or something similar).